app/src/main/java/org/mozilla/fenix/home/tabs/TabsAdapter.kt

```diff
@@ -44,43 +43,75 @@ class TabsAdapter(private val actionEmitter: Observer<TabsAction>) :
         }
     }
 
+    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int, payloads: MutableList<Any>) {
+        if (payloads.isEmpty()) onBindViewHolder(holder, position)
+        else if (holder is TabViewHolder) {
+            val o = payloads[0] as Bundle
+            for (key in o.keySet()) {
```

Comment:
Is it better to loop over keys than just pulling them out of the bundle?

```kotlin
o.getString(tab_url)?.apply(holder::updateUrl)
o.getBoolean(tab_selected)?.apply(holder::updateSelected)
```

If it is, we should probably replace the strings with the `tab_url` / `tab_selected` constants.